flask_admin/model/base.py

```diff
@@ -1477,6 +1477,10 @@ def create_view(self):
             return redirect(return_url)
 
         form = self.create_form()
+        if self._form_create_rules:
```

Comment:
How a form comes to be: `self.edit_form` -> `self._edit_form_class` -> `self._refresh_forms_cache` -> `self.get_edit_form` -> `self.get_form` -> `self.scaffold_form`.

Where would you say scaffolding stops and dynamic begins-- in `_refresh_forms_cache`?

The problem I see is every point of scaffolding code beyond `_refresh_forms_cache` is implemented or overridden by code outside the base class; so it would require changing all the contrib modules to implement it-- seems bad for compatibility.

Without doing this dynamically, if anyone has existing form overrides, they're still susceptible to this bug, and we have no room to warn them.

In order to maintain separation between form generation and form rendering rules, I think doing this dynamically is best: it allows us to prevent data loss due to this bug no matter how someone implements form generation.  How you set up your form rendering rules should not cause you to lose data.